src/OpenLoco/src/Vehicles/Vehicle.cpp

```diff
@@ -686,6 +686,90 @@ namespace OpenLoco::Vehicles
         return AirportObjectFlags::acceptsHeavyPlanes;
     }
 
+    // 0x004AFFF3
+    VehicleBogie* flipCar(VehicleBogie* frontBogie)
+    {
+        Vehicle train(frontBogie->head);
+        VehicleBase* precedingVehicleComponent = nullptr;
+        if (train.veh2->nextCarId == frontBogie->id)
+        {
+            precedingVehicleComponent = train.veh2;
+        }
+        CarComponent oldFirstComponent;
+        sfl::static_vector<CarComponent, VehicleObject::kMaxCarComponents> components;
+
+        for (auto& car : train.cars)
+        {
+            if (precedingVehicleComponent != nullptr)
+            {
+                oldFirstComponent = car;
+                for (CarComponent& component : car)
+                {
+                    std::swap(component.front->objectSpriteType, component.back->objectSpriteType);
+                    component.body->var_38 ^= Flags38::isReversed;
+                    components.push_back(component);
+                }
+                break;
+            }
+            if (car.back != nullptr && car.back->nextCarId == frontBogie->id)
```

Comment:
this doesn't look correct anymore. Its the body that would connect to the next car. I'm not sure moving this into this loop made it clearer.